pkg/kp/rollstore/consul_store.go

```diff
@@ -171,6 +171,20 @@ func (s consulStore) checkForConflictingUpdates(rcIDs rc_fields.IDs) error {
 	return nil
 }
 
+// Best-effort applies labels to the specified RC. It's the callers
+// responsibility to put things back the way they were in an error scenario due
+// to the variety of situations this is called in.
+func (s consulStore) labelNewRC(rcID rc_fields.ID, rcLabels klabels.Set) error {
+	for k, v := range rcLabels {
+		err := s.labeler.SetLabel(labels.RC, rcID.String(), k, v)
```

Comment:
I'm worried about this pattern - we need `labels.Applicator` to have a `SetLabels` method that can apply them atomically. In the near future, where we have active watchers for labels, we will want to guarantee atomicity when writing sets of related labels. Doing this would also reduce round-trips to Consul.